tensorflow-framework/src/test/java/org/tensorflow/framework/activations/SoftplusTest.java

```diff
@@ -0,0 +1,96 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+=======================================================================*/
+package org.tensorflow.framework.activations;
+
+import org.junit.jupiter.api.*;
+import org.tensorflow.Operand;
+import org.tensorflow.framework.utils.TestSession;
+import org.tensorflow.op.Ops;
+import org.tensorflow.types.TFloat32;
+import org.tensorflow.types.TFloat64;
+import org.tensorflow.types.TInt32;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+/** @author Jim Clarke */
+public class SoftplusTest {
+
+  private final TestSession.Mode[] tfModes = {TestSession.Mode.EAGER, TestSession.Mode.GRAPH};
+
+  public SoftplusTest() {}
+
+  @BeforeAll
+  public static void setUpClass() {}
+
+  @AfterAll
+  public static void tearDownClass() {}
+
+  @BeforeEach
+  public void setUp() {}
+
+  @AfterEach
+  public void tearDown() {}
+
+  /** Test of Softplus call method */
+  @Test
+  public void testCall__Int() {
+    int[] input = {1, -2, 3, -4, -1, 2, -3, 4};
+
+    for (TestSession.Mode tfMode : tfModes)
+      assertThrows(
+          java.lang.IllegalArgumentException.class,
+          () -> {
+            try (TestSession session = TestSession.createTestSession(tfMode)) {
+              Ops tf = session.getTF();
+              Softplus<TInt32> instance = new Softplus<>(tf);
+              Operand<TInt32> result = instance.call(tf.constant(input));
+            }
+          });
+  }
+
+  /** Test of Softplus call method */
+  @Test
+  public void testCall__Float() {
+    float[] input = {1, 2, 3, 4, 5, 6, 7, 8};
+    float[] expected = {
+      1.3132616F, 2.126928F, 3.0485873F, 4.01815F, 5.0067153F, 6.0024757F, 7.0009117F, 8.000336F
+    };
+    for (TestSession.Mode tfMode : tfModes)
+      try (TestSession session = TestSession.createTestSession(tfMode)) {
+        Ops tf = session.getTF();
+        Softplus<TFloat32> instance = new Softplus<>(tf);
+        Operand<TFloat32> result = instance.call(tf.constant(input));
+        session.evaluate(expected, result);
+      }
+  }
+
+  /** Test of Softplus call method */
+  @Test
+  public void testCall__Double() {
+    double[] input = {1, 2, 3, 4, 5, 6, 7, 8};
+    double[] expected = {
+      1.3132616875182228, 2.1269280110429727, 3.048587351573742,
```

Comment:
This should test that a negative or near zero value behaves appropriately rather than just the linear region.

Reply:
I have modified the input to use a combination of positive and negative values.